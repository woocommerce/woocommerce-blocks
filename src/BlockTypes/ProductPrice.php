<?php
namespace Automattic\WooCommerce\Blocks\BlockTypes;

/**
 * ProductPrice class.
 */
class ProductPrice extends AbstractBlock {

	/**
	 * Block name.
	 *
	 * @var string
	 */
	protected $block_name = 'product-price';

	/**
	 * API version name.
	 *
	 * @var string
	 */
	protected $api_version = '2';

	/**
	 * Get block supports. Shared with the frontend.
	 * IMPORTANT: If you change anything here, make sure to update the JS file too.
	 *
	 * @return array
	 */
	protected function get_block_type_supports() {
		return array(
			'color'                  =>
			array(
				'text'       => true,
				'background' => true,
				'link'       => false,
			),
			'typography'             =>
			array(
				'fontSize'                 => true,
				'__experimentalFontWeight' => true,
				'__experimentalFontStyle'  => true,
			),
			'__experimentalSelector' => '.wc-block-components-product-price',
		);
	}

}
