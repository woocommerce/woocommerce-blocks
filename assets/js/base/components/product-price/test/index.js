/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import ProductPrice from '../index';

describe( 'ProductPrice', () => {
	const currency = {
		code: 'GBP',
		currency_code: 'GBP',
		currency_decimal_separator: '.',
		currency_minor_unit: 2,
		currency_prefix: '£',
		currency_suffix: '',
		currency_symbol: '£',
		currency_thousand_separator: ',',
		decimalSeparator: '.',
		minorUnit: 2,
		prefix: '£',
		price: '61400',
		price_range: null,
		raw_prices: {
			precision: 6,
			price: '614000000',
			regular_price: '614000000',
			sale_price: '614000000',
		},
		regular_price: '61400',
		sale_price: '61400',
		suffix: '',
		symbol: '£',
		thousandSeparator: ',',
	};

	test( 'should use default price if no format is provided', () => {
		const component = TestRenderer.create(
			<ProductPrice
				price={ 50 }
				regularPrice={ 100 }
				currency={ currency }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should apply the format if one is provided', () => {
		const component = TestRenderer.create(
			<ProductPrice
				price={ 50 }
				regularPrice={ 100 }
				currency={ currency }
				format="pre price <price/> Test format"
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
