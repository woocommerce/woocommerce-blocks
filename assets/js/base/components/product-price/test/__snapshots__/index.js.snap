// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice should apply the format if one is provided 1`] = `
<span
  className="price wc-block-components-product-price"
>
  pre price 
  <span
    className="screen-reader-text"
  >
    Previous price:
  </span>
  <del
    className="wc-block-components-product-price__regular"
  >
    £1.00
  </del>
  <span
    className="screen-reader-text"
  >
    Discounted price:
  </span>
  <ins
    className="wc-block-components-product-price__value is-discounted"
  >
    £0.50
  </ins>
   Test format
</span>
`;

exports[`ProductPrice should use default price if no format is provided 1`] = `
<span
  className="price wc-block-components-product-price"
>
  <span
    className="screen-reader-text"
  >
    Previous price:
  </span>
  <del
    className="wc-block-components-product-price__regular"
  >
    £1.00
  </del>
  <span
    className="screen-reader-text"
  >
    Discounted price:
  </span>
  <ins
    className="wc-block-components-product-price__value is-discounted"
  >
    £0.50
  </ins>
</span>
`;
